cpp/examples/fetch_dependencies.cmake

```diff
@@ -19,12 +19,19 @@ file(
 )
 include(${CMAKE_BINARY_DIR}/cmake/get_cpm.cmake)
 
-set(CUDF_TAG branch-24.06)
-CPMFindPackage(
-  NAME cudf GIT_REPOSITORY https://github.com/rapidsai/cudf
-  GIT_TAG ${CUDF_TAG}
-  GIT_SHALLOW
-    TRUE
-    SOURCE_SUBDIR
-    cpp
-)
+# try to find an already built libcudf and use it.
+find_package(cudf HINTS ${cudf_ROOT}/latest)
+
+# build it via CPM
+if (NOT cudf_FOUND)
+  set(CUDF_TAG branch-24.06)
+
+  CPMFindPackage(
```

Comment:
I think CPMFindPackage can search for a matching local package. Is there something missing here that we need to make it discoverable? Do we need to add something like `VERSION 24.06.00` to the CPMFindPackage call?

Reply:
Good suggestion, I think CPM's default behavior is to download first but I think I can get around it and have it look locally first.